tests/integration/api_asset_test.py

```diff
@@ -244,3 +269,467 @@ def test_restore_path(
+    assert api_query_overload_2.name == API_QUERY_NAME
+    assert api_query_overload_2.connection_qualified_name == connection.qualified_name
+    assert api_query_overload_2.api_input_field_count == API_QUERY_INPUT_FIELD_COUNT
+    assert api_query_overload_2.api_query_output_type == API_QUERY_OUTPUT_TYPE
+    assert (
+        api_query_overload_2.api_query_output_type_secondary
+        == API_QUERY_OUTPUT_TYPE_SECONDARY
+    )
+    assert api_query_overload_2.connector_name == AtlanConnectorType.API.value
+
+
+@pytest.fixture(scope="module")
+def api_query_overload_3(
+    client: AtlanClient, connection: Connection
+) -> Generator[APIQuery, None, None]:
+    assert connection.qualified_name
+    to_create = APIQuery.creator(
+        name=API_QUERY_NAME,
+        connection_qualified_name=connection.qualified_name,
+        api_input_field_count=API_QUERY_INPUT_FIELD_COUNT,
+        api_query_output_type=API_QUERY_OUTPUT_TYPE,
+        api_query_output_type_secondary=API_QUERY_OUTPUT_TYPE_SECONDARY,
+        is_object_reference=API_QUERY_IS_OBJECT_REFERENCE,
+        reference_api_object_qualified_name=API_QUERY_REFERENCE_OBJECT_QUALIFIED_NAME,
+    )
+    response = client.asset.save(to_create)
+    result = response.assets_created(asset_type=APIQuery)[0]
+    yield result
+    delete_asset(client, guid=result.guid, asset_type=APIQuery)
+
+
+def test_api_query_overload_3(
+    client: AtlanClient, connection: Connection, api_query_overload_3: APIQuery
+):
+    assert api_query_overload_3
+    assert api_query_overload_3.guid
+    assert api_query_overload_3.qualified_name == API_QUERY_QUALIFIED_NAME
+    assert api_query_overload_3.name == API_QUERY_NAME
+    assert api_query_overload_3.connection_qualified_name == connection.qualified_name
+    assert api_query_overload_3.api_input_field_count == API_QUERY_INPUT_FIELD_COUNT
+    assert api_query_overload_3.api_query_output_type == API_QUERY_OUTPUT_TYPE
+    assert (
+        api_query_overload_3.api_query_output_type_secondary
+        == API_QUERY_OUTPUT_TYPE_SECONDARY
+    )
+    assert api_query_overload_3.api_is_object_reference == API_QUERY_IS_OBJECT_REFERENCE
+    assert (
+        api_query_overload_3.api_object_qualified_name
+        == API_QUERY_REFERENCE_OBJECT_QUALIFIED_NAME
+    )
+    assert api_query_overload_3.connector_name == AtlanConnectorType.API.value
+
+
+@pytest.fixture(scope="module")
+def api_field_parent_object(
+    client: AtlanClient, connection: Connection
+) -> Generator[APIField, None, None]:
+    assert connection.qualified_name
+    to_create = APIField.create(
+        name=API_FIELD_NAME,
+        parent_api_object_qualified_name=API_OBJECT_QUALIFIED_NAME,
+        parent_api_query_qualified_name=None,
+    )
+    response = client.asset.save(to_create)
+    result = response.assets_created(asset_type=APIField)[0]
+    yield result
+    delete_asset(client, guid=result.guid, asset_type=APIField)
```

Comment:
We should use `creator()` instead (ref: https://github.com/atlanhq/atlan-python/pull/403/files#r1800760792)